roles/common/tasks/main.yml

```diff
@@ -18,6 +18,10 @@
   tags:
     - common
 
+- name: ensure cloud.cfg.d directory exists
+  sudo: yes
+  file: path=/etc/cloud/cloud.cfg.d state=directory recurse=yes owner=root group=root mode=0644
```

Comment:
@siddharthist Okay, I've changed this commit to exclude vSphere as a provider rather than create the directory and tested that deployment works without error.  Anything else needed being able to pull it into master?  Want to get these back in before 1.1 is wrapped up.